fineract-provider/src/main/java/org/apache/fineract/infrastructure/event/business/domain/BulkBusinessEvent.java

```diff
@@ -30,4 +30,9 @@ public BulkBusinessEvent(List<BusinessEvent<?>> value) {
     public String getType() {
         return "BulkBusinessEvent";
```

Comment:
I know, a bit picky, but could we add somewhere `static final String` variables with these kind of values (if in doubt then put it in class `BulkBusinessEvent.java` as a private variable. Like it is now the string will always be created new when the function `getType()` is called. Having a variable is also a good practice in terms of avoiding "magic strings".

Another suggestion: why not use `BulkBusinessEvent.class.getSimpleName()` (="BulkBusinessEvent")? In that case this whole thing would be immune against refactoring errors...

If I were you I would do this:

```
public class BulkBusinessEvent imiplements BusinessEvent {
    private static final String TYPE = getClass().getSimpleName();

...

    @Override
    public String getType() {
        return TYPE;
    }

...

}
```

And finally: as the function `getType()` is required by all `BusinessEvent` classes you could create an abstract class and move above code there. Benefits:
- only have to write this code once
- completely type-safe
- avoiding all magic strings (aka refactoring works like a charm)
- ... and memory usage is a tiny bit better (no "new strings" on every call to `getType()`)

This could look like this:

```
public class AbstractBusinessEvent imiplements BusinessEvent {
    protected static final String TYPE = getClass().getSimpleName();

...

    @Override
    public String getType() {
        return TYPE;
    }

...

}

public class BulkBusinessEvent extends AbstractBusinessEvent {

...

}
```